src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -930,6 +930,32 @@ public static function provideSnake()
         ];
     }
 
+    /**
+     * @dataProvider provideUnderscore
+     */
+    public function testUnderscore(string $expectedString, string $origin): void
+    {
+        $instance = static::createFromString($origin)->underscore();
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideUnderscore(): array
```

Comment:
we must also have unicode tests for this method (not in AsciiTestCase of course)